parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -253,14 +253,21 @@ retrycmd_get_tarball_from_registry_with_oras() {
 retrycmd_get_access_token_for_oras() {
     retries=$1; wait_sleep=$2; url=$3
     for i in $(seq 1 $retries); do
-        ACCESS_TOKEN_OUTPUT=$(timeout 60 curl -v -s -H "Metadata:true" --noproxy "*" "$url")
-        if [ -n "$ACCESS_TOKEN_OUTPUT" ]; then 
+        response=$(timeout 60 curl -v -s -H "Metadata:true" --noproxy "*" "$url" -w "\n%{http_code}")
+        ACCESS_TOKEN_OUTPUT=$(echo "$response" | sed '$d')
+        http_code=$(echo "$response" | tail -n1)
+        if [ -n "$ACCESS_TOKEN_OUTPUT" ] && [ "$http_code" -eq 200 ]; then 
             echo "$ACCESS_TOKEN_OUTPUT"
             return 0
         fi
         sleep $wait_sleep
     done
-    return $ERR_ORAS_PULL_NETWORK_TIMEOUT
+    if [ -n "$http_code" ]; then
+        echo "failed to retrieve kubelet identity token from IMDS, http code: $http_code, msg: $ACCESS_TOKEN_OUTPUT"
+        return $ERR_ORAS_PULL_UNAUTHORIZED
```

Comment:
Would http_code != 200 always mean authorization issue? Not sure whether it would make sense to add more exit codes here, since we also log details about the error in line above

Reply:
this part is related to retrieve kubelet identity token to non-anonymous access to acr. Thus whatever the error is, it will result the cse unauthorized to access acr.

Not sure if we need more exit codes here. The sub code could be, as i observed: 
identity not found (400)
url is invalid (400)
imds connection timeout (imds endpoint issue)

I think we can leave one exit code for now to check feedback from customer/support to check if they need that many exit code. WDYT?

Reply:
Makes sense, let's leave this as is